pallets/parachain-staking/src/lib.rs

```diff
@@ -716,6 +727,98 @@ pub mod pallet {
+		/// - Writes: Round
+		/// # </weight>
+		#[pallet::weight(100_000)]
+		pub fn increase_max_candidate_stake(origin: OriginFor<T>, new: BalanceOf<T>) -> DispatchResult {
+			frame_system::ensure_root(origin)?;
+			ensure!(new >= T::MinCollatorCandidateStk::get(), Error::<T>::CannotSetBelowMin);
+			let old = <MaxCollatorCandidateStk<T>>::get();
+			ensure!(new > old, Error::<T>::NotIncreasing);
+			<MaxCollatorCandidateStk<T>>::put(new);
+
+			Self::deposit_event(Event::MaxCandidateStakeChanged(old, new));
+			Ok(())
+		}
+
+		/// Increases the maximum amount a collator candidate can stake. The new
+		/// maximum cannot be lower than MinCollatorCandidateStk.
+		///
+		/// The dispatch origin must be Root.
+		///
+		/// Emits `MaxCandidateStakeChanged`.
+		///
+		/// # <weight>
+		/// - The transaction's complexity is mainly dependent on updating the
+		///   `SelectedCandidates` storage in `select_top_candidates` which in
+		///   return depends on the number of `MaxSelectedCandidates` (N).
+		/// - For each N, we read `CollatorState` from the storage.
+		/// ---------
+		/// Weight: O(N) where N is `MaxSelectedCandidates` bounded by
+		/// `MaxCollatorCandidates`
+		/// - Reads: MaxCollatorCandidateStk, 2 * N * CollatorState,
+		///   CandidatePool
+		/// - Writes: MaxCollatorCandidateStk, N * CollatorState,
+		///   SelectedCandidates
+		/// # </weight>
+		#[pallet::weight(100_000)]
+		pub fn decrease_max_candidate_stake(origin: OriginFor<T>, new: BalanceOf<T>) -> DispatchResultWithPostInfo {
+			frame_system::ensure_root(origin)?;
+			ensure!(new >= T::MinCollatorCandidateStk::get(), Error::<T>::CannotSetBelowMin);
+
+			let old = <MaxCollatorCandidateStk<T>>::get();
+			ensure!(old > new, Error::<T>::NotDecreasing);
+
+			// iterate candidate pool and update all candidates with stake > max
+			let mut post_weight = T::DbWeight::get().reads(2);
+			<CandidatePool<T>>::mutate(|pool| {
+				let candidates = pool
+					.clone()
+					.into_vec()
+					.into_iter()
+					.map(|mut candidate| {
+						// state should always exist but let's be safe
+						if let Some(mut state) = <CollatorState<T>>::get(&candidate.owner) {
+							if state.stake > new {
+								// safe because we ensure stake > new
+								let delta = state.stake - new;
+								state.stake_less(delta);
+								candidate.amount = candidate.amount.saturating_sub(delta);
+
+								// immediately free delta by setting lock from old to new
+								T::Currency::set_lock(STAKING_ID, &candidate.owner, new, WithdrawReasons::all());
+								<CollatorState<T>>::insert(&candidate.owner, state);
+
+								// update consumed weight
+								post_weight = post_weight.saturating_add(T::DbWeight::get().writes(2));
+							}
+						}
+						candidate
+					})
+					.collect();
+				*pool = OrderedSet::from(candidates);
+			});
+
+			// update candidates for next round
+			let (_num_collators, _num_delegators, _, _) = Self::select_top_candidates();
```

Comment:
Do we even need to capture the return values here?

Reply:
Yes, for post weight reduction.